addons/voice/org.openhab.voice.googletts/src/main/java/org/openhab/voice/googletts/internal/cloudapi/GoogleTTSCloudImplementation.java

```diff
@@ -0,0 +1,259 @@
+
+            int status = ((HttpURLConnection) connection).getResponseCode();
+            if (HttpURLConnection.HTTP_OK != status) {
+                logger.error("Call {} returned HTTP {}", url, status);
+                throw new IOException("Could not read from service: HTTP code" + status);
+            }
+            if (logger.isTraceEnabled()) {
+                for (Entry<String, List<String>> header : connection.getHeaderFields().entrySet()) {
+                    logger.trace("Response.header: {}={}", header.getKey(), header.getValue());
+                }
+            }
+
+            if (is == null) {
+                is = connection.getInputStream();
+            } else {
+                is = new SequenceInputStream(is, connection.getInputStream());
+            }
+        }
+        return is;
+    }
+
+    /**
+     * Splits the text into multiple chunks that are processable by the Google Translate TTS API.
+     * It tries to split the text at sentence borders in order to improve intonation.
+     *
+     * @param text
+     *            The text to split
+     * @return The list of chunks that can be given to the Google Translate TTS API
+     */
+    private List<String> splitText(String text) {
+        String[] sentences = text.split("(?<=[.!?:;])");
+        List<String> chunks = new ArrayList<String>();
+        String currentChunk = "";
+
+        for (String sentence : sentences) {
+            if (!currentChunk.isEmpty() && (currentChunk.length() + sentence.length()) > MAX_CHARACTERS) {
+                chunks.add(currentChunk.trim());
+                currentChunk = "";
+            }
+            if (sentence.length() > MAX_CHARACTERS) {
+                chunks.addAll(splitSentence(sentence));
+            } else {
+                currentChunk += sentence;
+            }
+        }
+        if (!currentChunk.isEmpty()) {
+            chunks.add(currentChunk.trim());
+        }
+
+        return chunks;
+    }
+
+    /**
+     * Splits a sentence into multiple chunks that are processable by the Google Translate TTS API.
+     *
+     * @param text
+     *            The sentence to split
+     * @return The list of chunks that can be given to the Google Translate TTS API
+     */
+    private List<String> splitSentence(String sentence) {
+        String[] words = sentence.trim().split("(?<=[^\\w])");
+        List<String> chunks = new ArrayList<String>();
+        String currentChunk = "";
+
+        for (String word : words) {
+            if (!currentChunk.isEmpty() && currentChunk.length() + word.length() > MAX_CHARACTERS) {
+                chunks.add(currentChunk.trim());
+                currentChunk = "";
+            }
+            if (word.length() > MAX_CHARACTERS) {
+                logger.warn("Can not say '{}' because it exceeds the character limit of {} characters.", word,
+                        Integer.toString(MAX_CHARACTERS));
```

Comment:
You can shorten it to `logger.warn("Can not say '{}' because it exceeds the character limit of {} characters.", word, MAX_CHARACTERS)`. The logger takes care of the rest.